src/sql/workbench/services/accountManagement/browser/accountManagementService.ts

```diff
@@ -213,11 +213,25 @@ export class AccountManagementService implements IAccountManagementService {
 
 			let result = await self._accountStore.addOrUpdate(account);
 			if (result.accountAdded) {
+				// Double check that there isn't a matching account
+				if (result.changedAccount!.key.authLibrary === MSAL_AUTH_LIBRARY) {
+					let indexToRemove: number = provider.accounts.findIndex(account => {
+						// corner case handling for personal accounts
+						if (account.key.accountId.includes('#')) {
```

Comment:
Can we do an email comparison instead? 
An email account will always have single entry in the list of accounts right?

Reply:
Nvm, figured users do get different emails for same account.